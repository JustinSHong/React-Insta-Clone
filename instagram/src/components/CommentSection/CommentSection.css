.CommentSection_form {
	width: 1000px;
}

hr {
	background-color: black;
}

.CommentSection_input {
	width: 75%;
}
