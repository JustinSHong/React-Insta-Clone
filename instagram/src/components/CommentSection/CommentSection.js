import React from "react";
import moment from "moment";
import PropTypes from "prop-types";
// styles
import "./CommentSection.css";

class CommentSection extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			comments: this.props.commentList,
			username: "Guest",
			newComment: ""
		};
	}

	// handle new comments being typed in
	handleNewComment = event => {
		this.setState({ newComment: event.target.value });
		console.log(event.target.value);
	};

	// add new comments to a post's list of comments
	addNewComment = () => {
		const commentBox = this.state.comments;
		commentBox.push({
			username: this.state.username,
			text: this.state.newComment
		});
		this.setState({ comments: commentBox, newComment: "" });
	};

	render() {
		return (
			<div>
				{this.state.comments.map((comment, index) => (
					<div key={index}>
						<strong>{comment.username}</strong> {comment.text}
					</div>
				))}
				<div>
					<p className="timeStamp">
						{/* render timestamp of most recent comment made */}
						<span>{moment().calendar()}</span>
					</p>
					<hr />
					<form className="CommentSection_form">
						<input
							className="CommentSection_input border-top-0 border-right-0 border-left-0 border-bottom-0 py-4"
							type="text"
							name="newComment"
							value={this.state.newComment}
							onChange={this.handleNewComment}
							onClick={this.addNewComment}
							placeholder="Add a comment..."
						/>
					</form>
				</div>
			</div>
		);
	}
}

CommentSection.propTypes = {
	commentList: PropTypes.array.isRequired
};

export default CommentSection;
