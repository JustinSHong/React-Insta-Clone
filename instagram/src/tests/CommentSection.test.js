import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import moment from "moment";
// component
import CommentSection from "../components/CommentSection/CommentSection";
// data
import postData from "../dummy-data";

Enzyme.configure({ adapter: new Adapter() });

describe("<CommentSection />", () => {
    const tree = renderer.create(<CommentSection commentList={postData} />);
    it("should render correctly", () => {
        expect(tree).toMatchSnapshot();
    });

    it("should render a comment's author and text", () => {
        const component = shallow(<CommentSection commentList={postData} />);
        const username = component.find("strong");

        username.forEach((node, index) => {
            const user = node.props().children;
            expect(typeof user).toBe("string");

            for (let i = 0; i < postData[index].comments.length; i++) {
                const comment = postData[index].comments[i];
                expect(typeof comment).toBe("object");
                expect(typeof comment.username).toBe("string");
                expect(typeof comment.text).toBe("string");
                expect(comment.username).toBeTruthy();
                expect(comment.text).toBeTruthy();
            }
        });
    });

    it("should render a timestamp", () => {
        const component = shallow(<CommentSection commentList={postData} />);
        const timestamp = component.find("span");

        timestamp.forEach((node, index) => {
            const time = node.props().children;

            expect(typeof time).toBe("string");
            expect(typeof time).toBeTruthy();
            expect(time).toEqual(moment().calendar());
        });
    });

    it("should render an input tag", () => {
        const component = shallow(<CommentSection commentList={postData} />);
        const input = component.find(".CommentSection_input");

        expect(input.length).toBe(1);
        expect(input.props().name).toEqual("newComment");
        expect(input.props().type).toEqual("text");
    });

    it("should have initial state", () => {
        const data = tree.getInstance().state;
        const { comments, username, newComment } = data;

        expect(comments).toBeInstanceOf(Array);
        expect(typeof username).toBe("string");
        expect(username).toBeTruthy();
        expect(typeof newComment).toBe("string");
        expect(newComment).toBeFalsy();
    });

    it("should have an input field that changes value", () => {});

    it("should handle a new comment", () => {});

    it("should handle adding a new comment", () => {});
});
