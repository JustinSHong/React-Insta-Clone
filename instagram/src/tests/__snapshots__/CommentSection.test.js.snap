// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentSection /> should render correctly 1`] = `
<div>
  <div>
    <strong>
      philzcoffee
    </strong>
     
  </div>
  <div>
    <strong>
      twitch
    </strong>
     
  </div>
  <div>
    <strong>
      playhearthstone
    </strong>
     
  </div>
  <div>
    <p
      className="timeStamp"
    >
      <span>
        Today at 11:30 AM
      </span>
    </p>
    <hr />
    <form
      className="CommentSection_form"
    >
      <input
        className="CommentSection_input border-top-0 border-right-0 border-left-0 border-bottom-0 py-4"
        name="newComment"
        onChange={[Function]}
        onClick={[Function]}
        placeholder="Add a comment..."
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
